lib/jsdom/living/helpers/geometry/dommatrix-init.js

```diff
@@ -0,0 +1,89 @@
+"use strict";
+
+const { matrix2dArrayProperties, matrix3dArrayProperties } = require("./matrix");
+
+const aliasedProperties = [
+  ["a", "m11", 1],
+  ["b", "m12", 0],
+  ["c", "m21", 0],
+  ["d", "m22", 1],
+  ["e", "m41", 0],
+  ["f", "m42", 0]
+];
+
+// Properties that are required to be zero for a valid 2D Matrix
+const matrix2dZeroProps = [
+  "m13", "m14", "m23", "m24",
+  "m31", "m32", "m34", "m43"
+];
+
+// Properties that are required to be one for a valid 2D matrix
+const matrix2dOneProps = ["m33", "m44"];
+
+
+function hasProp(o, prop) {
+  return Object.prototype.hasOwnProperty.call(o, prop);
+}
+
+function propertyIsTrue(o, prop) {
+  return hasProp(o, prop) && Boolean(o[prop]);
+}
+
+// If both alias and actual properties are present, they must be equal.
+function validateAliasedProps(init) {
+  aliasedProperties.forEach(([alias, prop]) => {
+    if (
+      hasProp(init, alias) &&
+      hasProp(init, prop) &&
+      init[alias] !== init[prop]
```

Comment:
This uses [Strict Equality Comparison](https://tc39.es/ecma262/#sec-strict-equality-comparison), but the spec (https://drafts.fxtf.org/geometry/#matrix-validate-and-fixup-2d) calls for [SameValueZero](https://tc39.es/ecma262/#sec-samevaluezero) as the comparison algorithm. They have subtle differences w.r.t. the treatment of NaN, as documented at https://drafts.fxtf.org/geometry/#matrix-validate-and-fixup-2d:

> Note: The SameValueZero comparison algorithm returns true for two ‘NaN’ values, and also for ‘0’ and ‘-0’.

Make a new function called `sameValueZero` to handle this.